eessi/testsuite/tests/apps/osu.py

```diff
@@ -0,0 +1,296 @@
+The basic application class is taken from the hpctestlib to which extra features are added.
+"""
+
+import os
+import reframe as rfm
+from reframe.core.meta import parameters
+import reframe.utility.sanity as sn
+
+from hpctestlib.microbenchmarks.mpi.osu import osu_benchmark
+
+from eessi.testsuite import hooks, utils
+from eessi.testsuite.constants import SCALES, TAGS, DEVICE_TYPES
+
+def my_filtering_function():
+    """
+    Filtering function for filtering scales for the pt2pt OSU test
+    """
+    scale_filtered = SCALES.copy()
+    for key in list(SCALES):
+        if(key == '1_core' or key == '4_cores' or
+           SCALES.get(key).get('num_nodes') > 2):
+            scale_filtered.pop(key)
+        elif('node_part' in SCALES.get(key)):
+            if(SCALES.get(key).get('node_part') > 1):
+                scale_filtered.pop(key)
+    return scale_filtered
+
+
+def my_filtering_function_coll():
+    """
+    Filtering function for filtering scales for collective the OSU test
+    """
+    scale_filtered = SCALES.copy()
+    for key in list(SCALES):
+        if(key == '1_core'):
+            scale_filtered.pop(key)
+    return scale_filtered
+
+
+@rfm.simple_test
+class osu_pt_2_pt(osu_benchmark):
+    ''' Run-only OSU test '''
+    scale = parameter(my_filtering_function())
+    valid_prog_environs = ['default']
+    valid_systems = []
+    time_limit = '30m'
+    module_name = parameter(utils.find_modules('OSU-Micro-Benchmarks'))
+    # Device type for non-cuda OSU-Micro-Benchmarks should run on hosts of both
+    # node types. To do this the default device type is set to GPU.
+    device_type = DEVICE_TYPES['GPU']
+
+
+    @run_after('init')
+    def run_after_init(self):
+        """hooks to run after init phase"""
+        hooks.filter_valid_systems_by_device_type(
+               self,
+               required_device_type=self.device_type)
+        is_cuda_module = utils.is_cuda_required_module(self.module_name)
+        # This part of the hook is meant to be for the OSU cpu tests.
+        if not is_cuda_module and self.device_type == DEVICE_TYPES['GPU']:
+            self.valid_systems = ['*']
+            self.device_buffers = 'cpu'
+        elif is_cuda_module and self.device_type == DEVICE_TYPES['GPU']:
+            # Currently the device buffer is hard coded to be cuda. More
+            # options need to be introduced based on vendor and device type.
+            self.device_buffers = 'cuda'
+        # This part of the code removes the collective communication calls out
+        # of the run list since this test is only meant for pt2pt.
+        if not self.benchmark_info[0].startswith('mpi.pt2pt'):
+            self.valid_systems = []
+        hooks.set_modules(self)
+
+    @run_after('init')
+    def set_tag_ci(self):
+        if (self.benchmark_info[0] == 'mpi.pt2pt.osu_latency' or
+           self.benchmark_info[0] == 'mpi.pt2pt.osu_bw'):
+            self.tags.add('CI')
```

Comment:
```suggestion
        if (self.benchmark_info[0] in ['mpi.pt2pt.osu_latency', 'mpi.pt2pt.osu_bw']:
            self.tags.add('CI')
```

Reply:
There is one condition missing in the `hooks` if it is not a `cuda` module and the device type is `GPU`. With OSU non cuda module one can run `cpu` tests in the `gpu` partition.

Reply:
Incorporated.